tests/test_highscore_v2.py

```diff
@@ -0,0 +1,262 @@
+            "slayer": int,
+            "farming": int,
+            "runecraft": int,
+            "hunter": int,
+            "construction": int,
+            "league": int,
+            "bounty_hunter_hunter": int,
+            "bounty_hunter_rogue": int,
+            "cs_all": int,
+            "cs_beginner": int,
+            "cs_easy": int,
+            "cs_medium": int,
+            "cs_hard": int,
+            "cs_elite": int,
+            "cs_master": int,
+            "lms_rank": int,
+            "soul_wars_zeal": int,
+            "abyssal_sire": int,
+            "alchemical_hydra": int,
+            "barrows_chests": int,
+            "bryophyta": int,
+            "callisto": int,
+            "cerberus": int,
+            "chambers_of_xeric": int,
+            "chambers_of_xeric_challenge_mode": int,
+            "chaos_elemental": int,
+            "chaos_fanatic": int,
+            "commander_zilyana": int,
+            "corporeal_beast": int,
+            "crazy_archaeologist": int,
+            "dagannoth_prime": int,
+            "dagannoth_rex": int,
+            "dagannoth_supreme": int,
+            "deranged_archaeologist": int,
+            "general_graardor": int,
+            "giant_mole": int,
+            "grotesque_guardians": int,
+            "hespori": int,
+            "kalphite_queen": int,
+            "king_black_dragon": int,
+            "kraken": int,
+            "kreearra": int,
+            "kril_tsutsaroth": int,
+            "mimic": int,
+            "nex": int,
+            "nightmare": int,
+            "phosanis_nightmare": int,
+            "obor": int,
+            "phantom_muspah": int,
+            "sarachnis": int,
+            "scorpia": int,
+            "skotizo": int,
+            "tempoross": int,
+            "the_gauntlet": int,
+            "the_corrupted_gauntlet": int,
+            "theatre_of_blood": int,
+            "theatre_of_blood_hard": int,
+            "thermonuclear_smoke_devil": int,
+            "tombs_of_amascut": int,
+            "tombs_of_amascut_expert": int,
+            "tzkal_zuk": int,
+            "tztok_jad": int,
+            "venenatis": int,
+            "vetion": int,
+            "vorkath": int,
+            "wintertodt": int,
+            "zalcano": int,
+            "zulrah": int,
+            "rifts_closed": int,
+            "artio": int,
+            "calvarion": int,
+            "duke_sucellus": int,
+            "spindel": int,
+            "the_leviathan": int,
+            "the_whisperer": int,
+            "vardorvis": int,
+        }
+
+        # Check if the type of each value in the returned player dictionary matches the expected type
+        for key, expected_type in expected_types.items():
+            value = player.get(key)
+            # if value is not None:
+            assert isinstance(
+                value, expected_type
+            ), f"Key {key} has incorrect type. Expected: {expected_type}, Got: {type(value)}"
```

Comment:
if we are going this route for testing, which i'm not 100% convinced about, we should just have a pydantic class for this